### PR TITLE
[misc] fix: Support prompt logprobs in OpenAI server

### PR DESCRIPTION
**Pull request by yaoyu-33**
## Summary

The OpenAI-compatible server accepts prompt-logprob completion requests on its public HTTP surface, but its default engine configuration materialized only last-token logits. A request such as:

```json
{"model":"any","prompt":"Hello","max_tokens":1,"echo":true,"logprobs":1}
```

therefore reached MCore's collective request admission assertion instead of completing or returning a request-scoped error. The assertion escaped the serving loop, shut down the engine, and left subsequent clients unable to complete requests.

This change enables full-logit materialization by default for the request-driven server while retaining the existing `--return-log-probs` spelling as a compatibility flag.

## Root cause

`scripts/inference/openai_server.py` defaulted `return_log_probs` to false and passed it to the shared inference-config builder. That builder translated false to `materialize_only_last_token_logits=True`. The completions endpoint independently accepts `echo=true, logprobs=<integer>` and translates it to prompt-logprob sampling, which is incompatible with last-token-only materialization.

The fix is kept at the server ownership boundary. Shared synchronous and asynchronous offline generation defaults are unchanged.

## Regression evidence

Fail-before on unmodified `origin/main`:

```text
uv run --no-sync python -m pytest --confcutdir=tests/unit_tests/scripts \
  tests/unit_tests/scripts/test_openai_server_entrypoint.py -q
1 failed, 1 passed
```

The default launch propagated `return_log_probs=False`; the explicit-flag control passed.

Pass-after with the unchanged regression and adjacent async entrypoint test:

```text
uv run --no-sync python -m pytest --confcutdir=tests/unit_tests/scripts \
  tests/unit_tests/scripts/test_openai_server_entrypoint.py \
  tests/unit_tests/scripts/test_async_text_generation_entrypoint.py -q
3 passed
```

A real one-node GPU/server run also returned HTTP 200 with prompt log probabilities, kept the server alive, and returned HTTP 200 for an ordinary follow-up request; the original assertion was absent.

Additional checks:

```text
uv run --no-sync ruff check scripts/inference/openai_server.py \
  tests/unit_tests/scripts/test_openai_server_entrypoint.py
uv run --no-sync ruff format --check scripts/inference/openai_server.py \
  tests/unit_tests/scripts/test_openai_server_entrypoint.py
git diff --check
uv run --no-sync pre-commit run --all-files
```

All passed.

## Scope

This changes only the OpenAI-compatible server default and adds a CPU regression test. It does not change MCore, shared offline sampling defaults, model loading, dependencies, CI workflows, or public conversion APIs. Full-logit materialization can use more compute and memory than last-token-only generation; no performance, convergence, model-quality, or full-suite validation is claimed.

### CODE REVIEW

File: scripts/inference/openai_server.py [hunk 1]
```diff
@@ -74,7 +74,14 @@ def add_server_args(parser: argparse.ArgumentParser) -> argparse.ArgumentParser:
...
     )
-    group.add_argument("--return-log-probs", action="store_true", help="Materialize all logits for log probs.")
+    # The HTTP frontend accepts prompt-logprob requests per request. A last-token-only
+    # engine context asserts on those requests and tears down the serving loop.
+    group.add_argument(
+        "--return-log-probs",
+        action="store_true",
+        default=True,
+        help="Materialize all logits for request-level log probs (enabled for server compatibility).",
+    )
```
[the remark this answers is not in the document]
Comment (yaoyu-33) [reply scripts/inference/openai_server.py:84]: Addressed in 3963a4cf0: the option now uses argparse.BooleanOptionalAction, preserving --return-log-probs while adding --no-return-log-probs. The focused test covers default, explicit enable, and explicit disable.
